extend UnsplashBridge

langfingaz opened the pull request:
Thanks @nel50n for his UnsplashBridge! The commit is mainly based on his work.

As I this new bridge is based on `UnsplashBridge.php`, I left the `MAINTAINER` in line 6 to @nel50n . What do you think, shall I change this to me or leave it as is?

Signed-off-by: langfingaz <daniel@systemli.org>

langfingaz committed:
add Bridge: Image feed of Unsplash user

Thanks @nel50n for his UnsplashBridge! This Commit is mainly based on his work.

Signed-off-by: langfingaz <daniel@systemli.org>

langfingaz commented:
In this bridge one can "follow" a username on https://unsplash.com receiving new images.

Here is one example of a running rss-bridge instance with this bridge enabled: https://rss.privacy1st.de/?action=display&bridge=UnsplashUser&u=h4mmiidd&m=20&w=1920&q=75&format=Html

langfingaz commented:
And I'm not sure, but maybe one could merge this bridge into the already existing `UnsplashBridge` so that one can select if one would like to get the latest unsplas images or new images from a specific user. Help or ideas on how to do this are welcome!

theScrabi committed:
[NordbayernBridge] Fix images and newsblock order (#1741)

em92 commented:
> I left the `MAINTAINER` in line 6 to @nel50n . What do you think, shall I change this to me or leave it as is?

Generally speaking, if you are making new bridge, you should set MAINTAINER to yourself.

> And I'm not sure, but maybe one could merge this bridge into the already existing `UnsplashBridge`

You can add param like "Filter by Username (optional)" or "Username (optional)" and defaultValue = "". Then you can change api source depending if this param is given.

langfingaz committed:
extend UnsplashBridge:

+ filter photo stream for username
+ change image preview quality
+ add link to user, location (if present), link to photo on unsplash to $item['content']

Signed-off-by: langfingaz <daniel@systemli.org>

langfingaz commented:
> You can add param like "Filter by Username (optional)" or "Username (optional)" and defaultValue = "". Then you can change api source depending if this param is given.

All right, I did some more work on the UnsplashBridge and merged my new UnsplashUserBridge into the already existing UnsplashBridge.

> Generally speaking, if you are making new bridge, you should set MAINTAINER to yourself.

And I'm now added as second MAINTAINER.

---

You can test the modified UnsplashBridge here: [https://rss.privacy1st.de](https://rss.privacy1st.de/)

langfingaz committed:
Delete UnsplashUserBridge.php

Functionality is now merged into UnsplashBridge

VerifiedJoseph committed:
[SoundcloudBridge] Fix bridge not returning tracks (#1757)

+ Use artwork for enclosure

arnd-s committed:
[TwitterBridge] URL to js file with apikey changed again (#1764)

Ololbu committed:
[FicbookBridge] Fix data getting and months (#1765)

ORelio committed:
[NextINpact] Fix subtitle extraction in #LeBrief (#1780)

The bridge was taking another article abstract as subtitle for #LeBrief articles

triatic committed:
[TwitterBridge] Optimise regular expression code (#1768)

* [TwitterBridge] Optimise regular expression code

Optimise regular expression search code so adding new URLs is cleaner

corenting committed:
[NasaApodBridge] Fix header being parsed as item (#1586)

corenting committed:
[core] Use Parsedown for Markdown parsing (#1783)

csisoap committed:
[NasaApodBridge] Fix broken image link (#1778)

otakuf committed:
[AlbionOnlineBridge] New bridge (#1769)

VerifiedJoseph committed:
[HeiseBridge] Fix multi-page article fetching (#1767)

Fixes multi-page article fetching by adding '&seite=all' to  article URL.

Ololbu committed:
[FicbookBridge] Add getName implementation (#1771)

corenting committed:
[DribbbleBridge] Fix picture parsing (#1787)

kolarcz committed:
[CeskaTelevizeBridge] Add New bridge (#1784)

ORelio committed:
[ZDNet] Fix article layout (#1793)

em92 committed:
[.travis] Fix several phpcs and phpunit errors (#1799)

VerifiedJoseph committed:
[KoreusBridge + VarietyBridge] Use HTTPS when fetching feedburner feeds (#1797)

Commit:
[FacebookBridge] Handle mobile links and unify host validation (#1789)

gsantner committed:
[NineGagBridge] In post URI replace scheme from "http" to "https"

em92 committed:
[NineGagBridge] Lint previous commit

em92 committed:
[GoogleSearch] Use other class for content retreiving (#1803)

em92 commented:
Hi, @langfingaz ! Could you please rebase your PR?

I have recently fixed Travis CI. It runs some tests and linting checks.

ORelio committed:
[WordPress] Limit feed to 20 items (#1801)

langfingaz committed:
Merge pull request #2 from RSS-Bridge/master

update remote changes2

langfingaz commented:
Well, I'm sorry that I somehow failed rebasing my changes... The master branch of my forked repo is now 26 (and not 3) commits ahead of RSS-Bridge:master.

I created a branch `langfingaz/unsplash` which is 3 commits ahead of RSS-Bridge:master.
Probably it better to create a new pull request from within that branch and close this pull request to keep it cleaner?

Sorry for the inconvenience.
